osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs

```diff
@@ -103,29 +103,25 @@ private void load(OverlayColourProvider colours)
                             CornerRadius = CORNER_RADIUS,
                             Children = new Drawable[]
                             {
-                                new GridContainer
+                                new FillFlowContainer
                                 {
                                     RelativeSizeAxes = Axes.Both,
-                                    ColumnDimensions = new[]
-                                    {
-                                        new Dimension(GridSizeMode.Relative, 0.2f)
-                                    },
-                                    Content = new[]
+                                    Direction = FillDirection.Horizontal,
+                                    Children = new Drawable[]
                                     {
-                                        new Drawable[]
+                                        new Box
                                         {
-                                            new Box
-                                            {
-                                                RelativeSizeAxes = Axes.Both,
-                                                Colour = colours.Background5,
-                                            },
-                                            new Box
-                                            {
-                                                RelativeSizeAxes = Axes.Both,
-                                                Colour = ColourInfo.GradientHorizontal(colours.Background5, colours.Background5.Opacity(0.3f))
-                                            },
-                                        }
-                                    }
+                                            RelativeSizeAxes = Axes.Both,
+                                            Colour = colours.Background5,
+                                            Width = 0.2f,
+                                        },
+                                        new Box
+                                        {
+                                            RelativeSizeAxes = Axes.Both,
+                                            Colour = ColourInfo.GradientHorizontal(colours.Background5, colours.Background5.Opacity(0.3f)),
+                                            Width = 0.8f,
+                                        },
+                                    },
```

Comment:
You can take this one step further to reduce overhead even more:

```diff
diff --git a/osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs b/osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs
index 522438227a..f1fc751630 100644
--- a/osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs
+++ b/osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs
@@ -103,25 +103,19 @@ private void load(OverlayColourProvider colours)
                             CornerRadius = CORNER_RADIUS,
                             Children = new Drawable[]
                             {
-                                new FillFlowContainer
+                                new Box
                                 {
                                     RelativeSizeAxes = Axes.Both,
-                                    Direction = FillDirection.Horizontal,
-                                    Children = new Drawable[]
-                                    {
-                                        new Box
-                                        {
-                                            RelativeSizeAxes = Axes.Both,
-                                            Colour = colours.Background5,
-                                            Width = 0.2f,
-                                        },
-                                        new Box
-                                        {
-                                            RelativeSizeAxes = Axes.Both,
-                                            Colour = ColourInfo.GradientHorizontal(colours.Background5, colours.Background5.Opacity(0.3f)),
-                                            Width = 0.8f,
-                                        },
-                                    },
+                                    Colour = colours.Background5,
+                                    Width = 0.2f,
+                                },
+                                new Box
+                                {
+                                    Anchor = Anchor.TopRight,
+                                    Origin = Anchor.TopRight,
+                                    RelativeSizeAxes = Axes.Both,
+                                    Colour = ColourInfo.GradientHorizontal(colours.Background5, colours.Background5.Opacity(0.3f)),
+                                    Width = 0.8f,
                                 },
                                 new Container
                                 {

```

For the record, I don't know how much of an overhead difference there is betwen grid/flow container in this kind of simple usage (i think at least one level of nesting reduction). But reducing container nesting like my proposal will definitely help, removing any layout overheads.

These kinds of fixes can be very helpful in places with repeating drawables, so this is a good optimisation target (`DrawableRoom` is displayed in a long list).

Reply:
I was probably too fixated on the precedent, but if we want >2 boxes in the future (like the precedent), `Anchor` wouldn't work and would have to result in using `FillFlowContainer`.